libs/klv/src/main/java/org/codice/alliance/libs/klv/Utilities.java

```diff
@@ -25,4 +30,30 @@ public static boolean isEmptyString(Serializable serializable) {
     public static boolean isNotEmptyString(Serializable serializable) {
         return !isEmptyString(serializable);
     }
+
+    static void safelySetAttribute(Metacard metacard, Attribute attribute) {
+        safelySetAttribute(metacard, attribute.getName(), attribute.getValues());
+    }
+
+    static void safelySetAttribute(Metacard metacard, String attributeName, Serializable value) {
+        if (metacard.getMetacardType()
+                .getAttributeDescriptor(attributeName)
+                .getType()
+                .getBinding()
+                .isAssignableFrom(value.getClass())) {
+            metacard.setAttribute(new AttributeImpl(attributeName, value));
+        }
+    }
+
+    static void safelySetAttribute(Metacard metacard, String attributeName,
+            List<Serializable> valueList) {
+        if (valueList.stream()
+                .allMatch(serializable -> metacard.getMetacardType()
+                        .getAttributeDescriptor(attributeName)
```

Comment:
Same issue here w/r/t `getAttributeDescriptor()` potentially returning a null. Also, you could extract `metacard.getMetacardType.getAttributeDescriptor(attributeName).getType().getBinding()` so you're not reprocessing it for each element of the list. Won't save you much in the long run, but it is still more efficient.